src/extractor/guidance/intersection_handler.cpp

```diff
@@ -37,13 +38,41 @@ IntersectionHandler::IntersectionHandler(const util::NodeBasedDynamicGraph &node
 {
 }
 
+bool IntersectionHandler::SuppressModeNavigation(const EdgeID &via_edge,
+                                                 const EdgeID &connected,
+                                                 const SuppressModeListT &SUPPRESS_MODE_LIST) const
+{
+    const auto in_mode = node_based_graph.GetEdgeData(via_edge).travel_mode;
+    const auto out_mode = node_based_graph.GetEdgeData(connected).travel_mode;
+    return IntersectionHandler::SuppressModeNavigation(in_mode, out_mode, SUPPRESS_MODE_LIST);
+}
+
+bool IntersectionHandler::SuppressModeNavigation(const TravelMode &in_mode,
+                                                 const TravelMode &out_mode,
+                                                 const SuppressModeListT &SUPPRESS_MODE_LIST) const
+{
+    const auto suppress_in_mode = std::find(begin(SUPPRESS_MODE_LIST), end(SUPPRESS_MODE_LIST), in_mode);
+    if (suppress_in_mode != end(SUPPRESS_MODE_LIST))
+    {
```

Comment:
minor nitpick: the check for `in_mode == out_mode` is cheaper than the algorithm call. So doing it the other way around would be faster:
```
if( in_mode != out_mode )
 return false;

const auto suppress_in_mode = std::find(begin(SUPPRESS_MODE_LIST), end(SUPPRESS_MODE_LIST), in_mode);

return (suppress_in_mode != SUPPRESS_MODE_LIST.end();
```

In addition: using `end` like this requires the inclusion of `iterator` and a `using std::end`.
In this case, it's probably easier to simply call `SUPPRESS_MODE_LIST.end()`.